Document auto-managed migration pattern

millerdev opened the pull request:
@dannyroberts @ctsims

millerdev committed:
Document auto-managed migration pattern

emord commented:
Does this belong as a component of https://github.com/dimagi/commcare-hq/blob/master/docs/migrations.rst ?

also should this be added to the index/left sidebar https://github.com/dimagi/commcare-hq/blob/master/docs/index.rst?

millerdev committed:
Minor cleanup

millerdev committed:
Add links to toc and migrations page

millerdev commented:
@emord thanks for the feedback.

I briefly considered adding it to *migrations.rst*, but decided against it because I think making it a stand-alone document makes it easier to reference and read. Added in https://github.com/dimagi/commcare-hq/pull/23581/commits/87cc012ef3169ea2ec368dbb799d3c0419584907

I thought the index was automatically generated since I didn't see any reference to *migrations.rst* in there either. I see it is not. Added  in https://github.com/dimagi/commcare-hq/pull/23581/commits/87cc012ef3169ea2ec368dbb799d3c0419584907

emord reviewed (approved):
Awesome. thanks for documenting this.
